test/auth_plug_test.exs

```diff
@@ -116,4 +117,48 @@ defmodule AuthPlugTest do
 
     assert conn.assigns.person.email == "person@dwyl.com"
   end
+
+  describe "Test logout/1" do
+
+    setup %{endpoint: endpoint} do
+      test_conn =
+        conn(:get, endpoint)
+        |> init_test_session(%{})
+
+      {:ok, conn: test_conn}
+    end
+
+    @tag endpoint: "/admin"
+    test "logout", %{conn: conn} do
+      data = %{email: "alice@dwyl.com", name: "Alice"}
+      jwt = AuthPlug.Token.generate_jwt!(data)
+
+      conn =
+        conn
+        |> put_session(:jwt, jwt)
+        |> AuthPlug.call(@opts)
+
+
+      token = get_session(conn, :jwt)
+      {:ok, decoded} = AuthPlug.Token.verify_jwt(token)
+      assert Map.get(decoded, "email") == "alice@dwyl.com"
+
+      # logout:
+      conn_logged_out = AuthPlug.logout(conn)
+
+      # should no longer be logged in:
+      assert conn_logged_out.assigns == %{}
+
+      # attempt to access "/admin" endpoint
+      conn_end =
+        conn_logged_out
+        |> AuthPlug.call(@opts)
+
+      # should redirect as no longer authenticated/authorized:
+      assert conn_end.request_path == "/admin"
+      assert conn_end.status == 302
+
+      # IO.inspect(conn_end, label: "conn_end")
```

Comment:
We can remove this IO.inspect (I don't mind to leave it if you prefer to have it for future tests)

Reply:
yeah, I was still very much in the middle of trying to debug something in the tests ... 🙄 
will ping you once I've removed it. thanks for the reminder.